CMakeLists.txt

```diff
@@ -23,6 +23,9 @@ option(NATS_BUILD_STATIC_EXAMPLES "Statically link examples" OFF)
 option(NATS_BUILD_STREAMING "Build NATS Streaming" ON)
 option(NATS_BUILD_NO_PREFIX_CONNSTS "No prefix for connection status enum" OFF)
 option(NATS_COMPILER_HARDENING "Compiler hardening flags" OFF)
+option(NATS_BUILD_STATIC_LIBRARY "Build static library" ON)
```

Comment:
Could we make them `NATS_BUILD_LIB_STATIC` and `NATS_BUILD_LIB_SHARED`? This is not that important, but when doing `make edit_cache` this way they would be close to each other, that's all.